os/compression/compress_read.c

```diff
@@ -404,9 +402,20 @@ void compress_uninit(void)
 {
 #if CONFIG_COMPRESSION_TYPE == 1
 	/* Freeing memory allocated to read_buffer and out_buffer for file decompression */
-	if (compression_header.compression_format == COMPRESSION_TYPE_LZMA) {
-		kmm_free(buffers.read_buffer);
-		kmm_free(buffers.out_buffer);
+	if (compression_header->compression_format == COMPRESSION_TYPE_LZMA) {
+		if (buffers.read_buffer) {
+			kmm_free(buffers.read_buffer);
+			buffers.read_buffer = NULL;
+		}
+		if (buffers.out_buffer) {
+			kmm_free(buffers.out_buffer);
+			buffers.out_buffer = NULL;
+		}
 	}
 #endif
+
+	if (compression_header) {
```

Comment:
Is this conditional required? To make sure it is necessary, it should be null at some time. But if it is NULL, line 405 makes null dereference.

Reply:
Ok. 
Given that `compress_uninit` is only called if `compress_init` was success (i.e. kmm_malloc for compression_header was successful), I will remove this check.
It is `not needed`.

Reply:
Made this change.